### <font color= 'Red'>**Problem Statement:**</font>

We have a dataset containing Sachin Tendulkar's ODI cricket career stats, including various performance metrics and the outcomes of matches.

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [1]:
!wget https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7

--2024-09-02 02:49:32--  https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.google.com (drive.google.com)... 74.125.139.139, 74.125.139.138, 74.125.139.113, ...
Connecting to drive.google.com (drive.google.com)|74.125.139.139|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7 [following]
--2024-09-02 02:49:32--  https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.usercontent.google.com (drive.usercontent.google.com)... 108.177.12.132, 2607:f8b0:400c:c08::84
Connecting to drive.usercontent.google.com (drive.usercontent.google.com)|108.177.12.132|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 26440 (26K) [application/octet-stream]
Saving to: ‘uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7’

uc?id=1zBM3idCNWceB 100%[===================>]  25.82K  --.-KB/s    in 0s      

2024-09-0

In [4]:
df_sachin= pd.read_csv('uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7')
df_sachin.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [5]:
df_sachin.shape

(360, 14)

### <font color='purple'> **Q1. A match is randomly chosen, what is the probability that India have won that match?**</font>

In [13]:
ind_won = df_sachin.loc[df_sachin['Won']==True]

In [16]:
len(ind_won)

184

In [17]:
len(df_sachin)

360

The probabitly that India have won the match is,

= $\Large\frac{Number\ of\ matches\ india\ won}{Numnber \ of\ matches\ india\ played}$

In [20]:
Winning_prob= len(ind_won)/len(df_sachin)
Winning_prob

0.5111111111111111

<font color= 'orange'> Conclusion: </font>

If a match is randomly chosen, there is **51%** chance that India have won that match.

### <font color='purple'>  **Q2. A match is chosen at a random, what is the probability that Sachin has scored a Century in that match?** </font>

In [22]:
sachin_century= df_sachin.loc[df_sachin['century']==True]

In [23]:
len(sachin_century)

46

In [25]:
len(df_sachin)

360

The probabilty of sachin scoreing a century is:

= $\Large\frac{Number\ of\ centuries\ scored}{Number \ of \ matches\ played}$

In [28]:
len(sachin_century)/len(df_sachin)

0.12777777777777777

<font color= 'Orange'> Conclusion:<font>

  If you chose a random match, there is **12.77% chance** that Sachin has scored a century in that match

### <font color= 'Purple'>**Cross tab or Contingency table:**</font>

Using cross table or continegency table we can determing all posible conditional and marginal probabilties.

In [30]:
pd.crosstab(index= df_sachin['century'],columns=df_sachin['Won'],margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

Using `pd.crosstab()` converted the output into nice tabular format

- **Century** is taken as the **index** and **Won** is taken as **columns**    

-   When we do **Margins = True** we get **All**, both in rows and columns,

  - The values of **All** in a ROW represnts the **Total Value** of each columns (False, True, All)  

  - The values of **All** in a COLUMN represents the **Total Value** of each rows (False, True, All)

This table is also known as <font color='purple'> **`Contingency Table`** </font>

We can calculate probabilities using the contingency table.  

### <font color='purple'> **Q3. A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India have won that match?**</font>

Using the above crosstab, we can determine the answer to this question,

In [32]:
# prob of winning and century
# Won -> True, century -> True

30/360

0.08333333333333333

<font color='orange'> Conclusion : </font>
  
  There is **8% chance** that Sachin has scored a century and India have won that match if we choose a random match


This tells us, that **contingency table** is more convenient to calculate probabilities rather than hard coded the every single line

### <font color='purple'>**Conclusion of the Problem statement:** </font>    

Let's have a look how is Sachin's batting can or cannot impact the winning chances of India  


1. Out of the <font color='purple'>**360** matches</font> that Sachin has played, <font color='purple'>India have **won 184** matches and Loose 176 matches. </font>

2. So, if we choose any match at a random from Sachin's ODI career, there is a <font color='purple'>**51%** chance that India have won that match.     </font>


3. Now, If we choose a random match from Sachin's ODI career, there is <font color='purple'>**12.77%** chance that Sachin has scored a century in that match.  </font>

4. We know if a random match is choosen, there is 12.77% chance that Sachin has scored a century but  

  there is <font color='purple'>**only 8%** chance India have won that match.    </font>

  - we can conclude that the <font color='orange'>chances of India, Winning a match is more when Sachin didn't score a century </font> (what an amazing insight)





<br>

Finally,

We can conclude that, if we pick a random match where Sachin played, India's win percentage is 51%. There is 12.77% chance of Sachin scoring a century in that match, and there is only 8% chance that in that match Sachin scores a century as well as India have won that match


<br><br>